crates/stackable-operator/src/builder/pod/mod.rs

```diff
@@ -50,9 +51,21 @@ pub enum Error {
 
     #[snafu(display("object is missing key {key:?}"))]
     MissingObjectKey { key: &'static str },
+
+    #[snafu(display("The volume {volume_name:?} is clashing with an already existing volume with the same name but a different content. \
+        The existing volume is {existing_volume:?}, the new one is {new_volume:?}"))]
+    ClashingVolumeName {
+        volume_name: String,
+        existing_volume: Box<Volume>,
+        new_volume: Box<Volume>,
+    },
 }
 
 /// A builder to build [`Pod`] or [`PodTemplateSpec`] objects.
+///
+/// This struct is often times using an [`IndexMap`] to have consistent ordering (so we don't produce reconcile loops).
+/// We are also choosing it over an [`BTreeMap`], as it's easier to debug for users, as logically grouped volumes
+/// (e.g. all volumes related to S3) are near each other in the list instead of "just" being sorted alphabetically.
```

Comment:
```suggestion
/// This struct is often times using an [`IndexMap`] to have consistent ordering (so we don't produce reconcile loops).
/// We are also choosing it over a [`BTreeMap`], because it is easier to debug for users, as logically grouped volumes
/// (e.g. all volumes related to S3) are near each other in the list instead of "just" being sorted alphabetically.
```